htlcswitch/interfaces.go

```diff
@@ -144,6 +142,19 @@ type ChannelUpdateHandler interface {
 	// htlc to the channel, taking the amount of the htlc to add as a
 	// parameter.
 	MayAddOutgoingHtlc(lnwire.MilliSatoshi) error
+}
+
+// ChannelUpdateHandler is an interface that provides methods that allow
+// sending lnwire.Message to the underlying link as well as querying state.
+type ChannelUpdateHandler interface {
+	// HandleChannelUpdate handles the htlc requests as settle/add/fail
+	// which sent to us from remote peer we have a channel with.
+	//
+	// NOTE: This function MUST be non-blocking (or block as little as
```

Comment:
either it is not blocking or it is not, can you explain why this function might be blocking ?